rfcs/convergence/file-naming.md

```diff
@@ -0,0 +1,152 @@
+| Example.tsx         | example.ts          | `export const Example = () => {..}`                                                 | To enforce rendering logic only within `.view.` type file/s. TS will enforce this by default. Not extremely picky on this one, we can use `.tsx` as well to match extension with test/story - priority is to be consistent                                                                                                |
+| Example.test.tsx    | example.spec.tsx    | `describe->it`                                                                      | Integration test on JSDOM level (jest) for whole component                                                                                                                                                                                                                                                                |
+| storybook (NONE)    | example.stories.tsx | `export const Example = () => {..}`                                                 | Specific file type for Storybook consumption. Main "starting" point for inner loop - daily development.                                                                                                                                                                                                                   |
+| mdx (NONE)          | example.stories.mdx | ---                                                                                 | Markdown(MDX) for consumer facing documentation(mainly for storybook at the moment, but can be transformed to any other tool that can parse MDX).                                                                                                                                                                         |
+| renderExample.tsx   | example.view.tsx    | `export const render = () => {..}`                                                  | `.render` might work as well, to match symbol name.                                                                                                                                                                                                                                                                       |
+| useExample.ts       | example.hook.ts     | `export const useExample = () => {}`                                                | main logic encapsulated under hook -> name should be the same as encapsulated component (in our case `example`)                                                                                                                                                                                                           |
```

Comment:
This breaks common convention in React community:
- https://github.com/adobe/react-spectrum/tree/main/packages/%40react-spectrum/utils/src
- https://github.com/reakit/reakit/tree/master/packages/reakit-utils/src
- https://github.com/mui-org/material-ui/tree/next/packages/material-ui-utils/src
- https://github.com/streamich/react-use/tree/master/src

Reply:
I'd just add that there are no existing conventions ratified in any kind of official community style-guide, nor in fluent ui org :)  that's the primary aim of this RFC -> focus/ratification on consistency.

Reply:
That's fair, I used these repos as a reference to show a common pattern. Fluent's repo follows it, too, currently.
Another fair note will be, that there is an explicit naming convention for hooks, they should start with `use*()` keyword.

https://github.com/facebook/react/blob/483358c38f8623358dd44a3e57476bc02e6357bc/packages/eslint-plugin-react-hooks/src/RulesOfHooks.js#L17-L19